packages/components/src/unit-control/utils.ts

```diff
@@ -210,18 +205,23 @@ export function parseUnit( initialValue, units = ALL_CSS_UNITS ) {
  * Parses a number and unit from a value. Validates parsed value, using fallback
  * value if invalid.
  *
- * @param {number|string}            next          The next value.
- * @param {Array<WPUnitControlUnit>} units         Units to derive from.
- * @param {number|string}            fallbackValue The fallback value.
- * @param {string}                   fallbackUnit  The fallback value.
- * @return {Array<number, string>} The extracted number and unit.
+ * @param  next          The next value.
+ * @param  units         Units to derive from.
+ * @param  fallbackValue The fallback value.
+ * @param  fallbackUnit  The fallback value.
+ * @return The extracted number and unit.
  */
-export function getValidParsedUnit( next, units, fallbackValue, fallbackUnit ) {
+export function getValidParsedUnit(
+	next: number | string,
+	units: Array< WPUnitControlUnit > | false,
+	fallbackValue: number | string,
+	fallbackUnit: string
+) {
 	const [ parsedValue, parsedUnit ] = parseUnit( next, units );
 	let baseValue = parsedValue;
-	let baseUnit;
+	let baseUnit: string;
 
-	if ( isNaN( parsedValue ) || parsedValue === '' ) {
+	if ( ! Number.isFinite( parsedValue ) || parsedValue === '' ) {
```

Comment:
We should consider this change carefully, since it introduces a runtime change — `isNaN` doesn't always return the same results as `! Number.isFinite`

Reply:
Thanks for pointing this out, I meant to leave a specific comment for it. With the switch to TypeScript, using `isNaN` here generated an error:

![image](https://user-images.githubusercontent.com/14988353/135031441-6c4c17a4-b268-4972-95cc-e91bfb5e0eb6.png)

This seems to be that in TS, `isNaN` expects to be called with a variable of type `number`, but our `parsedValue` is either a number or a string. At this point in `getValidParsedUnit` the `parsedValue` variable must be either a number or an empty string (the results of calling `parseUnit`). So, I think `Number.isFinite` is a more appropriate check than `isNaN` anyway, at least from my mental model of what this function should be doing:

* Check that the number is not `NaN`
* Check that the number is a valid value that we can use (e.g. not `Infinity` or `-Infinity`) and not an arbitrary string

If it doesn't meet the above criteria then we fall back to the `fallbackValue`. One difference in using `Number.isFinite` is that the value won't be coerced to a number before checking to see if it's finite. I _think_ this is okay, because the value has already been coerced in `parseUnit` so at this point it should be a real (finite) value in order for us to consider it valid.

Please let me know if you think that's not right, though! Happy to change the logic here, I think it's the main place where I had to introduce a runtime change, but 🤞 it isn't an unexpected one in usage.

It could be that we might want to tighten up the Value type a bit further, too, given that in some places it'll only be `number` or the string literal `''` — I was a bit cautious about doing that, as we'd then have two different Value types floating around in the file, which could get confusing? But it also might make it a bit clearer 🤔

Reply:
> Please let me know if you think that's not right, though! Happy to change the logic here, I think it's the main place where I had to introduce a runtime change, but 🤞 it isn't an unexpected one in usage.

Thank you for the explanation! With this context (and the extra reassurance given by the passing unit tests), I think we can leave your new implementation untouched.

I would maybe only add a comment with a short version of the explanation you just gave in your previous message.

> It could be that we might want to tighten up the Value type a bit further, too, given that in some places it'll only be number or the string literal '' — I was a bit cautious about doing that, as we'd then have two different Value types floating around in the file, which could get confusing? But it also might make it a bit clearer 🤔

I had the same thought and came to the same conclusion